### PR TITLE
HPCC-22013 Replace website-antlr3 submodule with libantlr3c

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@mckellyln Please review

File: esp/services/ws_sql/antlr3c.cmake [hunk 1]
```diff
@@ -41,7 +41,7 @@ endif()
 include(ExternalProject)
 ExternalProject_Add(
     antlr3c
-    DOWNLOAD_COMMAND tar xzf ${CMAKE_CURRENT_SOURCE_DIR}/website-antlr3/download/C/libantlr3c-3.4.tar.gz
+    DOWNLOAD_COMMAND tar xzf ${CMAKE_CURRENT_SOURCE_DIR}/libantlr3c/libantlr3c-3.4.tar.gz
```
Comment (mckellyln) [inline esp/services/ws_sql/antlr3c.cmake]: It is a little convoluted that the build untars a tarfile with the same files as what is already in the source.  But this is not really any different than it was.
The:
build/esp/services/ws_sql/antlr3c/src/libantlr3c-3.4/src
dir created from the tar xf cmd, and the:
HPCC-Platform/esp/services/ws_sql/libantlr3c/src
are identical.  Same for include dir and m4 dir.

**Comment by mckellyln** [comment]
@Michael-Gardner I am ok with this change.  It is better than before by having the submodule only have the files we use, and it works correctly.  But I think it may be worth another look, as it seems the submodule source from the tarfile appears to match the files already in the source dir.

**Comment by richardkchapman** [comment]
I think there's a reason we do it this way (but I can't remember what it is!)

**Comment by mckellyln** [comment]
Perhaps then we don't need the source in the source dir ?
Something to look at, but as a separate issue ?
But I am good with this PR as is, it is good and an improvement over the previous.

**Comment by Michael-Gardner** [comment]
@mckellyln Originally I had thought I needed to include the source as is, with the license file, but I also wanted to include the archive to retain the same logic that we were using in the ExternalProject_Add's DOWNLOAD_COMMAND.  I could probably just change the DOWNLOAD_COMMAND to cp that entire directory over into the build directory, and then remove the archive.  I'll do that tomorrow and get back to you.

**Comment by mckellyln** [comment]
@Michael-Gardner back to you one more time.

**Comment by Michael-Gardner** [comment]
@mckellyln I modified the libantlr3c repository and removed that redundant jar.  I altered the DOWNLOAD step on add external project to do a copy instead of an extraction.  Had to make other small changes to insure that it worked correctly, and could find all the necessary files, but nothing major.

**Comment by richardkchapman** [comment]
7.4 would be a more appropriate target

**Comment by Michael-Gardner** [comment]
Rebasing it now

**Comment by mckellyln** [comment]
I see it has already been merged.

**Pull request merged**